cognite/neat/_graph/transformers/_value_type.py

```diff
@@ -64,3 +71,73 @@ def transform(self, graph: Graph) -> None:
                     graph.remove((s, property_uri, o))
                     new_property = URIRef(f"{property_uri}_{remove_namespace_from_uri(value_type_uri)}")
                     graph.add((s, new_property, o))
+
+
+class ConvertLiteral(BaseTransformer):
+    description: str = "ConvertLiteral is a transformer that converts a literal value."
+    _use_only_once: bool = False
+    _need_changes = frozenset({})
+
+    _count_by_properties = """SELECT (COUNT(?property) AS ?propertyCount)
+    WHERE {{
+      ?instance a <{subject_type}> .
+      ?instance <{subject_predicate}> ?property
+    }}"""
+
+    _properties = """SELECT ?instance ?property
+    WHERE {{
+      ?instance a <{subject_type}> .
+      ?instance <{subject_predicate}> ?property
+    }}"""
+
+    def __init__(
+        self,
+        subject_type: URIRef,
+        subject_predicate: URIRef,
+        conversion: Callable[[Any], Any],
+    ) -> None:
+        self.subject_type = subject_type
+        self.subject_predicate = subject_predicate
+        self.conversion = conversion
+        self._type_name = remove_namespace_from_uri(subject_type)
+        self._property_name = remove_namespace_from_uri(subject_predicate)
+
+    def transform(self, graph: Graph) -> None:
+        count_query = self._count_by_properties.format(
+            subject_type=self.subject_type, subject_predicate=self.subject_predicate
+        )
+
+        property_count_res = list(graph.query(count_query))
+        property_count = int(property_count_res[0][0])  # type: ignore [index, arg-type]
+        iterate_query = self._properties.format(
+            subject_type=self.subject_type, subject_predicate=self.subject_predicate
+        )
+
+        for instance, literal in iterate_progress_bar(  # type: ignore[misc]
+            graph.query(iterate_query),
+            total=property_count,
+            description=f"Converting {self._type_name}.{self._property_name}.",
+        ):
+            if not isinstance(literal, rdflib.Literal):
```

Comment:
you do not need this if when you introduce:
```
...
      FILTER(isIRI(?value))
```

Reply:
That is a nice learning. A few things, first shouldn't it be `Filter(isLiteral(?value))`. 

Second, is the user select a property which is a connection, this filter will silently ignore it which can lead to confusion. I think the current solution has the weakness that it will spam warnings for every case which is wrong. I suggest that I do a check first to see if there are any URI for the property, and then issue a warning for that? Wdyt?